src/main/java/io/apicurio/datamodels/cmd/commands/CommandFactory.java

```diff
@@ -591,9 +594,26 @@ public static final ICommand createChangeParameterDefinitionTypeCommand(Document
         throw new RuntimeException("Document type not supported by this command.");
     }
 
+    /**
+     * @deprecated Prefer the {@link CommandFactory#createChangePropertyTypeCommand(DocumentType, IPropertySchema, SimplifiedPropertyType)} 
+     * alternative that handles non Oas document types
+     */
+    @Deprecated
     public static final ICommand createChangePropertyTypeCommand(IPropertySchema property,
-            SimplifiedPropertyType newType) {
-        return new ChangePropertyTypeCommand(property, newType);
+                                                                 SimplifiedPropertyType newType) {
+        return new ChangePropertyTypeCommand_Oas(property, newType);
+    }
+    
+    public static final ICommand createChangePropertyTypeCommand(DocumentType docType,
+                                                                 IPropertySchema property,
+                                                                 SimplifiedPropertyType newType) {
+        if (docType == DocumentType.openapi2 || docType == DocumentType.openapi3) {
+            return new ChangePropertyTypeCommand_Oas(property, newType);
+        }
+        if (docType == DocumentType.asyncapi2) {
+            return new ChangePropertyTypeCommand_Aai20(property, newType);
+        }
+        throw new RuntimeException("Document type not supported by this command.");
```

Comment:
This is fine, although you can get the `DocumentType` from the property schema.  Just case it to a `Node` and then get the owner document and then the type.  Something like:

```java
DocumentType dt = ((Node) property).ownerDocument().getDocumentType();
```

Might make things easier vs. deprecating the method and creating a new one?

Reply:
Indeed, I didn't think of that one.